src/proxy/ssh/AgentForwarding.ts

```diff
@@ -0,0 +1,280 @@
+          console.log('[LazyAgent] Closed temporary agent channel after getIdentities');
+        }
+
+        callback(null, keys);
+      } catch (err: any) {
+        console.error('[LazyAgent] Error getting identities:', err);
+        if (agentProxy) {
+          agentProxy.close();
+        }
+        callback(err);
+      }
+    }).catch((err) => {
+      console.error('[LazyAgent] Unexpected error in executeWithLock:', err);
+      callback(err);
+    });
+  }
+
+  /**
+   * Sign data with a specific key using the client's forwarded agent
+   */
+  sign(
+    pubKey: any,
+    data: Buffer,
+    options: any,
+    callback?: (err: Error | null, signature?: Buffer) => void,
+  ): void {
+    if (typeof options === 'function') {
+      callback = options;
+      options = undefined;
+    }
+
+    if (!callback) {
+      callback = () => {};
+    }
+
+    console.log('[LazyAgent] sign called');
+
+    // Wrap the operation in a lock to prevent concurrent channel usage
+    this.executeWithLock(async () => {
+      console.log('[LazyAgent] Lock acquired, opening temporary channel for signing');
+      let agentProxy: SSHAgentProxy | null = null;
+
+      try {
+        agentProxy = await this.openChannelFn(this.client);
+        if (!agentProxy) {
+          throw new Error('Could not open agent channel');
+        }
+        let pubKeyBlob: Buffer;
+
+        if (typeof pubKey.getPublicSSH === 'function') {
+          pubKeyBlob = pubKey.getPublicSSH();
+        } else if (Buffer.isBuffer(pubKey)) {
+          pubKeyBlob = pubKey;
+        } else {
+          console.error('[LazyAgent] Unknown pubKey format:', Object.keys(pubKey || {}));
+          throw new Error('Invalid pubKey format - cannot extract SSH wire format');
+        }
+
+        const signature = await agentProxy.sign(pubKeyBlob, data);
+        console.log(`[LazyAgent] Signature received (${signature.length} bytes)`);
+
+        if (agentProxy) {
+          agentProxy.close();
+          console.log('[LazyAgent] Closed temporary agent channel after sign');
+        }
+
+        callback!(null, signature);
+      } catch (err: any) {
+        console.error('[LazyAgent] Error signing data:', err);
+        if (agentProxy) {
+          agentProxy.close();
+        }
+        callback!(err);
+      }
+    }).catch((err) => {
+      console.error('[LazyAgent] Unexpected error in executeWithLock:', err);
+      callback!(err);
+    });
+  }
+}
+
+/**
+ * Open a temporary agent channel to communicate with the client's forwarded agent
+ * This channel is used for a single request and then closed
+ *
+ * IMPORTANT: This function manipulates ssh2 internals (_protocol, _chanMgr, _handlers)
+ * because ssh2 does not expose a public API for opening agent channels from server side.
+ *
+ * @param client - The SSH client connection with agent forwarding enabled
+ * @returns Promise resolving to an SSHAgentProxy or null if failed
+ */
+export async function openTemporaryAgentChannel(
```

Comment:
How are we managing closing the open channels? Wouldn't this keep opening channels as long as users keep making requests?